sensor/common/filesystem/pipeline/pipeline.go

```diff
@@ -0,0 +1,174 @@
+			Path:     fs.GetRename().GetNew().GetPath(),
+			HostPath: fs.GetRename().GetNew().GetHostPath(),
+		}
+		activity.Operation = storage.FileActivity_RENAME
+	case *sensorAPI.FileActivity_Permission:
+		activity.File = &storage.FileActivity_File{
+			Path:     fs.GetPermission().GetActivity().GetPath(),
+			HostPath: fs.GetPermission().GetActivity().GetHostPath(),
+			Meta: &storage.FileActivity_FileMetadata{
+				Mode: fs.GetPermission().GetMode(),
+			},
+		}
+		activity.Operation = storage.FileActivity_PERMISSION_CHANGE
+	case *sensorAPI.FileActivity_Ownership:
+		activity.File = &storage.FileActivity_File{
+			Path:     fs.GetOwnership().GetActivity().GetPath(),
+			HostPath: fs.GetOwnership().GetActivity().GetHostPath(),
+			Meta: &storage.FileActivity_FileMetadata{
+				Uid:      fs.GetOwnership().GetUid(),
+				Gid:      fs.GetOwnership().GetGid(),
+				Username: fs.GetOwnership().GetUsername(),
+				Group:    fs.GetOwnership().GetGroup(),
+			},
+		}
+		activity.Operation = storage.FileActivity_OWNERSHIP_CHANGE
+	case *sensorAPI.FileActivity_Write:
+		activity.File = &storage.FileActivity_File{
+			Path:     fs.GetWrite().GetActivity().GetPath(),
+			HostPath: fs.GetWrite().GetActivity().GetHostPath(),
+		}
+		activity.Operation = storage.FileActivity_WRITE
+	case *sensorAPI.FileActivity_Open:
+		activity.File = &storage.FileActivity_File{
+			Path:     fs.GetOpen().GetActivity().GetPath(),
+			HostPath: fs.GetOpen().GetActivity().GetHostPath(),
+		}
+		activity.Operation = storage.FileActivity_OPEN
+	default:
+		log.Warn("Not implemented file activity type")
+		return nil
+	}
+
+	return activity
+}
+
+func (p *Pipeline) getIndicator(process *sensorAPI.ProcessSignal) *storage.ProcessIndicator {
+	pi := &storage.ProcessIndicator{
+		Id: uuid.NewV4().String(),
+		Signal: &storage.ProcessSignal{
+			Id:           process.GetId(),
+			Uid:          process.GetUid(),
+			Gid:          process.GetGid(),
+			Time:         process.GetCreationTime(),
+			Name:         process.GetName(),
+			Args:         process.GetArgs(),
+			ExecFilePath: process.GetExecFilePath(),
+			Pid:          process.GetPid(),
+			Scraped:      process.GetScraped(),
+			ContainerId:  process.GetContainerId(),
+		},
+	}
+
+	if process.GetContainerId() != "" {
+		// TODO(ROX-30798): Enrich file system events with deployment details
+		metadata, ok, _ := p.clusterEntities.LookupByContainerID(process.GetContainerId())
+		if !ok {
+			// unexpected - process should exist before file activity is
+			// reported
+			log.Debug("Container ID:", process.GetContainerId(), "not found for file activity")
+		} else {
+			pi.DeploymentId = metadata.DeploymentID
+			pi.ContainerName = metadata.ContainerName
+			pi.PodId = metadata.PodID
+			pi.PodUid = metadata.PodUID
+			pi.Namespace = metadata.Namespace
+			pi.ContainerStartTime = protocompat.ConvertTimeToTimestampOrNil(metadata.StartTime)
+			pi.ImageId = metadata.ImageID
+		}
+	}
+	// TODO: populate node info otherwise
```

Comment:
I've added an issue to track this https://issues.redhat.com/browse/ROX-31434 


```suggestion
	// TODO(ROX-31434): populate node info otherwise
```